doc/cprover-manual/restrict-function-pointer.md

```diff
@@ -0,0 +1,173 @@
+[CPROVER Manual TOC](../../)
+
+## Restricting function pointers
+
+### Motivation
+
+CBMC comes with a way to resolve calls to function pointers to direct function
```

Comment:
The documentation needs to be updated to reference `goto-instrument` instead of `cbmc` (also usage examples need to be updated to emphasise that this is a two-four step process source →binary→instrument→cbmc